pkg/datastore.go

```diff
@@ -105,9 +116,18 @@ func (s DefaultDatastore) GetHeaderByBlockHash(ctx context.Context, bh types.Has
 		return HeaderAndTrace{}, err
 	}
 
-	var trace HeaderAndTrace
-	err = json.Unmarshal(data, &trace)
-	return trace, err
+	var headers []HeaderAndTrace
+	if err = json.Unmarshal(data, &headers); err != nil {
+		return HeaderAndTrace{}, err
+	}
+
+	for _, header := range headers {
+		if strings.Compare(header.Header.BlockHash.String(), bh.String()) == 0 {
```

Comment:
This can be simplified to `if header.Header.BlockHash == bh`.  We'll get a small performance win, too.

Reply:
what about equalFold 👀👀 https://dzone.com/articles/optimizing-string-comparisons-in-go

Reply:
I don't think we care so much about the performance win (my fault for bringing it up 😆) as the legibility. 

Also, I saw a bunch of similar string comparisons littered around the codebase.   We should definitely fix those as well.

Reply:
Thank you for the review gents. But many of these comparisons are optimized in the refactor data API PR. So after merging that this will be fixed

Reply:
LGTM then!